cdap-ui/app/cdap/components/DataPrep/SamplerDropdown/index.js

```diff
@@ -0,0 +1,139 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, {Component} from 'react';
+import {UncontrolledDropdown} from 'components/UncontrolledComponents';
+import {DropdownToggle, DropdownMenu, DropdownItem} from 'reactstrap';
+import IconSVG from 'components/IconSVG';
+import DataPrepStore from 'components/DataPrep/store';
+import {objectQuery} from 'services/helpers';
+import T from 'i18n-react';
+import classnames from 'classnames';
+import MyDataPrepApi from 'api/dataprep';
+import NamespaceStore from 'services/NamespaceStore';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import {setWorkspace} from 'components/DataPrep/store/DataPrepActionCreator';
+require('./SamplerDropdown.scss');
+const samplerMethods = {
+  first: 'first',
+  bernoulli: 'bernoulli',
+  poisson: 'poisson',
+  reservoir: 'reservoir'
+};
+const PREFIX = `features.DataPrep.SamplerDropdown`;
+const NUMLINES = 10000;
+const DEFAULT_FRACTION = 0.35;
+const CONTENT_TYPE = 'text/plain';
+
+export default class SamplerDropdown extends Component {
+  constructor(props) {
+    super(props);
+    let samplerMethod = objectQuery(DataPrepStore.getState(), 'dataprep', 'workspaceInfo', 'properties', 'sampler');
+    this.dropdownOptions = Object.keys(samplerMethods).map(method => ({
+      name: method,
+      label: T.translate(`${PREFIX}.${method}.label`)
+    }));
+    let defaultValue = this.dropdownOptions.filter(option => option.name === samplerMethod);
+    this.state = {
+      samplerMethod: defaultValue.length ? defaultValue[0] : this.dropdownOptions[0]
+    };
+    this.onSamplerChange = this.onSamplerChange.bind(this);
+  }
+  componentDidMount() {
+    this.storeSubscription = DataPrepStore.subscribe(() => {
+      let samplerMethod = objectQuery(DataPrepStore.getState(), 'dataprep', 'workspaceInfo', 'properties', 'sampler');
+      let match = this.dropdownOptions.filter(method => method.name === samplerMethod);
+      if (match.length && match[0].name !== this.state.samplerMethod.name) {
+        this.setState({
+          samplerMethod: match[0]
+        });
+      }
+    });
+  }
+  onSamplerChange(samplerMethod) {
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    let path = objectQuery(DataPrepStore.getState(), 'dataprep', 'workspaceInfo', 'properties', 'path');
+    let workspaceId = objectQuery(DataPrepStore.getState(), 'dataprep', 'workspaceId');
+
+    let params = {
+      namespace,
+      path,
+      lines: NUMLINES,
+      fraction: DEFAULT_FRACTION,
+      sampler: samplerMethod.name
+    };
+    const headers = {
+      'Content-Type': CONTENT_TYPE
```

Comment:
So couple of things,
1. Right now that seems to be the only content type we wrangle
2. UI doesn't have that information in workspaceInfo right now. Is the assumption that we pick it up from file name? @nitinmotgi 

Noticed the same `readFile` call from file browser and backend returns `text/plain` as 'content.type` when we transition from File Browser to DataPrep for all files (as far as I have checked)